Packs/CommonScripts/Scripts/ParseEmailFilesV2/ParseEmailFilesV2_test.py

```diff
@@ -417,3 +419,83 @@ def test_md_output_with_body_text():
 
     md = data_to_md(email_data)
     assert expected == md
+
+
+@pytest.mark.parametrize('nesting_level_to_return, output, res', [('All files', ['output1'], ['output1']),
+                                                                  ('Outer file', ['output1', 'output2', 'output3'],
+                                                                   ['output1']),
+                                                                  ('Inner file', ['output1', 'output2', 'output3'],
+                                                                   ['output3'])])
+def test_parse_nesting_level(nesting_level_to_return, output, res):
+    """
+    Given:
+    - parsed email output, nesting_level_to_return param - All files.
+    - parsed email output, nesting_level_to_return param - Outer file.
+    - parsed email output, nesting_level_to_return param - Inner file.
+    When:
+    - Getting all nested emails.
+    - Getting only outer email file.
+    - Getting only inner email file.
+    Then: Validate that returned result as expected.
+    """
+    assert parse_nesting_level(nesting_level_to_return, output) == res
+
+
+@pytest.mark.parametrize('nesting_level_to_return, results_len, depth, results_index', [('All files', 4, 0, 2),
+                                                                                        ('Outer file', 3, 0, 2),
+                                                                                        ('Inner file', 1, 1, 0)])
+def test_eml_contains_eml_nesting_level(mocker, nesting_level_to_return, results_len, depth, results_index):
+    """
+    Given:
+    - A eml file contains eml, nesting_level_to_return param - All files.
+    - A eml file contains eml, nesting_level_to_return param - Outer file.
+    - A eml file contains eml, nesting_level_to_return param - Inner file.
+    When:
+    - Getting all nested emails.
+    - Getting only outer email file.
+    - Getting only inner email file.
+    Then: Validate that returned result as expected.
```

Comment:
Also here the given and when are twisted.
Also, the then should be more specific to each scenario for example: `Validating the that length of the response is x and the max size is y.`